### PR TITLE
doc: nrf_802154: corrections of the doc

### PR DESCRIPTION
**Pull request by ankuns**
Removed mentions of nRF52811
Replaced whole chapter "PA/LNA module (FEM support)" which was outdated and replaced with short section in "Driver features" referencing MPSL.
Removed outdated notes stating that some features are not supported on nRF53.

Signed-off-by: Andrzej Kuros <andrzej.kuros@nordicsemi.no>

### CODE REVIEW

**Review by jciupis — APPROVED** [review]
That's a much needed improvement, thanks!

**Review by dawidprzybylo — APPROVED** [review]

**Review by ahasztag — APPROVED** [review]

**Review by FrancescoSer — APPROVED** [review]

File: nrf_802154/doc/feature_description.rst [hunk 1]
```diff
@@ -361,3 +357,10 @@ The higher layer can implement various retransmission schemes by combining the i
    If the higher layer marks a frame as already secured, it must not expect the driver to update the dynamic data of the frame.
    Transmitting an encrypted frame with its header modified afterward results in a security breach.
    An attempt to transmit a frame with such parameters will fail unconditionally.
+
+Supporting Front-End Modules
+****************************
+
+The driver supports Front-End Modules (FEMs) using an implementation provided by the `mpsl`_.
```
Comment (anangl) [inline nrf_802154/doc/feature_description.rst:364]: Documentation build [fails on this line](https://github.com/nrfconnect/sdk-nrf/runs/6589279046?check_suite_focus=true) with the following error:
```
/home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrfxlib/src/nrf_802154/doc/feature_description.rst:364:Unknown target name: "mpsl".
```

**Review by anangl — CHANGES_REQUESTED** [review]

File: nrf_802154/doc/feature_description.rst (hunk 1 above)
Comment (ankuns) [reply nrf_802154/doc/feature_description.rst:364]: pushed fix already

Comment (ankuns) [reply nrf_802154/doc/feature_description.rst]: @anangl doc built correctly

**Review by anangl — APPROVED** [review]

**Pull request merged**